### PR TITLE
Issue #14456 - Enable search in timezone select component

### LINKED ISSUE #14456: Enable search for timezone selection widget
In #14429 we now use the system timezone list to populate the timezone select on the client -- as mentioned in #8943 people might have several hundred timezones on their system (I have about 600 locally for example), so we should enable the search option for the TZ widget. Tom mentioned that it's already an option so hopefully it should be straightforward.

### PR DESCRIPTION
**Pull request by daltojohnso**
Resolves #14456 -- just needed to pass the prop down to the Select component.

### CODE REVIEW

**Comment by ariya** [comment]
Excellent! Probably an opportunity to add a new Cypress test for this?

**Comment by daltojohnso** [comment]
Good idea -- will do.

**Comment by daltojohnso** [comment]
@ariya LMK if you find my added Cypress test satisfactory. I wasn't sure if there is a side effect in our analytics code I can look for like in the formatting test above the one I added.

**Comment by ariya** [comment]
> @ariya LMK if you find my added Cypress test satisfactory. I wasn't sure if there is a side effect in our analytics code I can look for like in the formatting test above the one I added.

Looks really good!

@nemanjaglumac do you fancy taking a pass at it?

**Comment by nemanjaglumac** [comment]
> @nemanjaglumac do you fancy taking a pass at it?

I will. Thanks for tagging me.

File: frontend/test/metabase/scenarios/admin/settings/settings.cy.spec.js [hunk 1]
```diff
@@ -178,6 +178,27 @@ describe("scenarios > admin > settings", () => {
...
+      .closest("li")
+      .find(".AdminSelect")
+      .click();
+
+    cy.focused().type("Centr");
+    cy.focused()
+      .closest(".MB-Select")
+      .find("a")
+      .last()
+      .click({ force: true });
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/admin/settings/settings.cy.spec.js:196]:
```suggestion
    cy.findByText("US/Central").click({ force: true });
```
We don't know if order is going to change, or if it depends on the context. Using the "last" link can backfire in the future.
This is much more explicit and it saves us a couple of lines :)

File: frontend/test/metabase/scenarios/admin/settings/settings.cy.spec.js [hunk 2]
```diff
@@ -178,6 +178,27 @@ describe("scenarios > admin > settings", () => {
...
+    cy.server();
+    cy.route("PUT", "**/report-timezone").as("reportTimezone");
+
+    cy.visit("/admin/settings/localization");
+    cy.contains("Report Timezone")
+      .closest("li")
+      .find(".AdminSelect")
+      .click();
+
+    cy.focused().type("Centr");
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/admin/settings/settings.cy.spec.js:191]:
```suggestion
    cy.findByPlaceholderText("Find...").type("Centr");
```
There are really helpful addons for Cypress ([`@testing-library/cypress`](https://github.com/testing-library/cypress-testing-library)) such as this one.

`cy.focused` was ok for this and will probably keep working, but what if that input doesn't receive focus in the future (for whatever reason)?

Another alternative would be:
```javascript
popover().within(() => {
  cy.get("input").type("Centr");
});
```

But you can see how much more elegant `cy.findByPlaceholderText()` is :)

**Comment by nemanjaglumac** [comment]
Looks and works great! 🎉

I didn't want to block you by "requesting a change", but I'd highly suggest implementing a few simple changes I suggested. They should make this test more resilient and easier to read.

**Comment by nemanjaglumac** [comment]
Another hint/suggestion (not as important as the previous ones, but it could be useful for other tests as well). If we use Cypress [alias](https://docs.cypress.io/api/commands/as.html#Syntax), we can reuse it later in the test. Additionally, we can make sure that "US/Central" isn't found anywhere in the page arbitrarily (`cy.contains("US/Central")` will not fail even if it finds multiple instances - beware). It wouldn't hurt to make sure it's within `.AdminSelect` specifically.

Aliasing helps us in both cases:
```javascript
// cy.contains("Report Timezone") - again, there could be multiple instances of the string on the page
cy.findByText("Report Timezone")
  .closest("li")
  .find(".AdminSelect")
  .as("selectMenu") // Aliasing this exact element
  .click();
cy.findByPlaceholderText("Find...").type("Centr");
cy.findByText("US/Central").click({ force: true });

cy.wait("@reportTimezone");
// Reusing the alias, and then asserting on it
cy.get("@selectMenu").contains("US/Central");
```

Another nice thing about the aliases is that Cypress marks them in the Cypress test runner:
image

I hope you find this useful.

**Comment by daltojohnso** [comment]
Very helpful, thank you @nemanjaglumac!

**Review by ariya — APPROVED** [review]
LGTM :+1:

**Pull request merged**